cmake/kokkos_arch.cmake

```diff
@@ -406,6 +406,13 @@ FOREACH (_BACKEND OPENMP PTHREAD QTHREAD HPX)
     SET(_HOST_PARALLEL ${_BACKEND})
   ENDIF()
 ENDFOREACH()
+
+IF(NOT _HOST_PARALLEL AND NOT KOKKOS_ENABLE_SERIAL)
+  MESSAGE(FATAL_ERROR "At least one host execution space must be enabled, "
+                      "but no host parallel execution space was requested "
+                      "and Kokkos_ENABLE_SERIAL=OFF.")
+ENDIF()
```

Comment:
@masterleinad Not sure about that one.  I believe the intended behavior was that SERIAL get's enabled if no host execution space was enabled.  @jjwilke can you confirm?

Reply:
My understanding was that the desired behavior from Christian and Jeff as that if NOTHING gets enabled, then SERIAL gets enabled by default. Otherwise SERIAL is not enabled if anything else is enabled.  Also, we only want to ensure that AT LEAST 1 gets enabled, but it's okay if more than one is enabled.

@crtrott: is there a "specification" of the desired configure behavior on this?

Reply:
If no host execution space is enabled SERIAL should be on. We need one host execution space.